c++/src/kj/async.h

```diff
@@ -63,6 +63,49 @@ using PromiseForResult = _::ReducePromises<_::ReturnType<Func, T>>;
 // T.  If T is void, then the promise is for the result of calling Func with no arguments.  If
 // Func itself returns a promise, the promises are joined, so you never get Promise<Promise<T>>.
 
+// =======================================================================================
+
+class AsyncObject {
+  // You may optionally inherit privately from this to indicate that the type is a KJ async object,
```

Comment:
Any reason for inheritance versus composition? I.e.,
```cpp
class IoThingy {
  kj::AsyncObject asyncDestructorGuard;
};
```
I like composition for this use case because, at least to me, it reinforces the optional nature of the debugging aid.

Reply:
Yes: Inheritance of an empty class adds zero bytes to the derived class's size. Composition adds at least one byte and up to 8 depending on padding.